engine/src/faction-boards/types.ts

```diff
@@ -88,8 +94,27 @@ export class FactionBoard {
   brainstone: BrainstoneArea;
   handlers?: { [event: string]: (player: Player, ...args: any[]) => any };
 
-  constructor(input: FactionBoardRaw) {
-    Object.assign(this, merge({}, defaultBoard, input));
+  constructor(input: FactionBoardVariants, customization: FactionCustomization) {
+    let data = merge({}, defaultBoard, input.standard);
+    if (customization != null) {
+      //not present in cloning
+      const abilities = customization.abilities;
+      const variants = input.variants;
+      if (abilities != "standard" && variants != null) {
+        const byPlayerCount = variants.find(
+          (v) => v.type == customization.abilities && v.players == customization.players
+        );
+        if (byPlayerCount != null) {
+          data = merge(data, byPlayerCount.board);
+        } else {
+          const byType = variants.find((v) => v.type == customization.abilities);
+          if (byType != null) {
+            data = merge(data, byType.board);
+          }
+        }
+      }
+    }
```

Comment:
Maybe make a function that returns the variant to merge (just `{}` if nothing found) and use that function here:

```
const boardCustomization = factionBoardCustomization(faction, customization)
let data = merge({}, defaultBoard, input.standard, boardCustomization)
```

And then move the `boardCustomization = ...` to `player.loadFaction` itself as per my first comment